### PR TITLE
Fast auth iframe wallet

### PR DESCRIPTION
**Pull request by esaminu**
Update fast auth to use an iframe / redirect redirect wallet selector module instead of embedded

### CODE REVIEW

File: src/lib/selector/fast-auth-wallet.ts [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+          method: 'POST',
+          mode: 'cors',
+          body: JSON.stringify(Array.from(encodeSignedDelegate(signedDelegate))),
+          headers: new Headers({ 'Content-Type': 'application/json' }),
+        });
+      }),
+    );
+    return true;
+  }
+  return false;
```
Comment (hcho112) [inline src/lib/selector/fast-auth-wallet.ts:58]: this function includes events for posting request to relayer which is async, but it doesn't return the promise and instead it simply returns boolean without tracking the promise. Would it be nicer to return the promise?

Also should we be checking error handling when signed delegate failed on relayer? (which has happened multiple times already)

File: src/lib/selector/fast-auth-wallet.ts [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+          blockHeightTtl: 60,
+          receiverId: receiverId as string,
+        });
+
+        await fetch(params.relayerUrl, {
+          method: 'POST',
+          mode: 'cors',
+          body: JSON.stringify(Array.from(encodeSignedDelegate(signedDelegate))),
+          headers: new Headers({ 'Content-Type': 'application/json' }),
+        });
```
Comment (hcho112) [inline src/lib/selector/fast-auth-wallet.ts:232]: error handling here too

File: src/lib/selector/fast-auth-wallet.ts [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+            blockHeightTtl: 60,
+            receiverId: receiverId as string,
+          });
+
+          await fetch(params.relayerUrl, {
+            method: 'POST',
+            mode: 'cors',
+            body: JSON.stringify(Array.from(encodeSignedDelegate(signedDelegate))),
+            headers: new Headers({ 'Content-Type': 'application/json' }),
+          });
```
Comment (hcho112) [inline src/lib/selector/fast-auth-wallet.ts:270]: maybe not in this PR, but I see repeated code regards to sending to relayer, perhaps we could refactor this into a function and just pass body perhaps?

File: src/lib/selector/fastAuthWalletConnection.ts [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+                'No window found in context, please ensure you are using WalletConnection on the browser',
+              );
+            };
+          }
+          return target[property as keyof FastAuthWalletConnection];
+        },
+      });
+    }
+    const iframe = document.createElement('iframe');
+    iframe.allow = 'publickey-credentials-create *; publickey-credentials-get *; clipboard-write';
```
Comment (hcho112) [inline src/lib/selector/fastAuthWalletConnection.ts:93]: I remember that browser console was complaining about `publickey-credentials-create *;` is invalid. You sure this is where it should be at?

**Review by hcho112 — COMMENTED** [review]
Left few comments but these are non-blockers.

File: src/lib/selector/fast-auth-wallet.ts (hunk 1 above)
Comment (esaminu) [reply src/lib/selector/fast-auth-wallet.ts]: Good catch, removed this function and handled the listener in `FastAuthWalletConnection`. Re error handling, I'm not sure how the VM handles those but I will address it in a separate PR as they are not currently handled for existing wallets

File: src/lib/selector/fast-auth-wallet.ts (hunk 2 above)
Comment (esaminu) [reply src/lib/selector/fast-auth-wallet.ts:209]: Per above these aren't currently addressed in prod and the caller here is the VM, will address in a follow up PR

File: src/lib/selector/fastAuthWalletConnection.ts (hunk 4 above)
Comment (esaminu) [reply src/lib/selector/fastAuthWalletConnection.ts]: Good catch, removed from now - this was added to the spec but hasn't been implemented by most modern browsers yet https://github.com/w3c/webauthn/issues/1656#issuecomment-1528147812

**Review by hcho112 — APPROVED** [review]
LGTM

**Pull request merged**